src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -148,6 +148,71 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
             });
     }
 
+#pragma warning disable ASPIREINTERACTION001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+    /// <summary>
+    /// Sets the description of the parameter resource.
+    /// </summary>
+    /// <param name="builder">Resource builder for the parameter.</param>
+    /// <param name="description">The parameter description.</param>
+    /// <returns>Resource builder for the parameter.</returns>
+    public static IResourceBuilder<ParameterResource> WithDescription(this IResourceBuilder<ParameterResource> builder, string description)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(description);
+
+        builder.Resource.Description = description;
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Sets the description of the parameter resource and configures it to use markdown.
+    /// </summary>
+    /// <param name="builder">Resource builder for the parameter.</param>
+    /// <param name="description">The parameter markdown description.</param>
+    /// <returns>Resource builder for the parameter.</returns>
+    public static IResourceBuilder<ParameterResource> WithMarkdownDescription(this IResourceBuilder<ParameterResource> builder, string description)
```

Comment:
nit: What about setting this using an optional parameter on `WithDescription`?

```cs
parameterBuilder.WithDescription("This is **great**!", enableDescriptionMarkdown: true);
```

Reply:
Yeah I went back and forward on that and landed on this, but I'm not strongly attached to it. Happy to change.